### PR TITLE
Updated go.md

### PR DESCRIPTION
**Pull request by MakMukhi**
@menghanl

### CODE REVIEW

File: docs/quickstart/go.md [hunk 1]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+---
+
+<p class="lead">This guide gets you started with gRPC in Go with a simple
+working example.</p>
+
+<div id="toc"></div>
+
+## Before you begin
+
+### Prerequisites
```
Comment (menghanl) [inline docs/quickstart/go.md]: Maybe we should specify go1.5 or higher here.

File: docs/quickstart/go.md [hunk 2]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+
+1. Install the protoc compiler that is used to generate gRPC service code. The simplest way to do this is to download pre-compiled binaries for your platform here: https://github.com/google/protobuf/releases
+
+2. Next, install the protoc plugin for Go by following the instructions here: https://github.com/golang/protobuf#installation	
+  *  ``` $ go get -u github.com/golang/protobuf/{proto,protoc-gen-go} ```
+  *  The compiler plugin, protoc-gen-go, will be installed in $GOBIN, defaulting to $GOPATH/bin. It must be in your $PATH for the protocol compiler, protoc, to find it.
+     
+    ``` $ export PATH=$PATH:$GOPATH/bin ```
+
+##Download the example
```
Comment (menghanl) [inline docs/quickstart/go.md]: Add a space after `##` here.

File: docs/quickstart/go.md [hunk 3]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+The grpc code that was fetched with ``` $ go get google.golang.org/grpc ``` also contains the examples. They can be found under the examples dir; $GOPATH/src/google.golang.org/grpc/examples
```
Comment (menghanl) [inline docs/quickstart/go.md]: `;` should be `:`?
Back-quote the path.
Add `.` at the end to make it a complete sentence here and other places where it's missing.

File: docs/quickstart/go.md [hunk 4]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+
+## Build the example
+Change to the example directory
+
+```$ cd $GOPATH/src/google.golang.org/grpc/examples/helloworld```
+
+For gRPC services to work we want to compile the proto file to generate a pb.go file that contains the generated code which is then used by the server and client apps.
```
Comment (menghanl) [inline docs/quickstart/go.md]: The generated `pb.go` file for this example is shipped with the code. So this step is actually not necessary.
Should we mention this?

File: docs/quickstart/go.md [hunk 5]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+
+For gRPC services to work we want to compile the proto file to generate a pb.go file that contains the generated code which is then used by the server and client apps. 
+
+To generate pb.go file run the following command (using protoc compiler):
+
+```$ protoc -I helloworld/ helloworld/helloworld.proto --go_out=plugins=grpc:helloworld```
+
+This generates `helloworld.pb.go`, which contains our generated client and server code, as well as code for populating, serializing, and retrieving our `HelloRequest` and `HelloResponse` message types.
+
+Most failures at this point are a result of a faulty installation (or having
```
Comment (menghanl) [inline docs/quickstart/go.md]: What are the failures here?
It seems to me if errors happen here, they should be caused by faulty installation of `protoc` or proto go plugin, not gRPC...

File: docs/quickstart/go.md [hunk 6]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+This regenerates the helloworld.pb.go with our new changes.
+
+## Update and run the application
+
+We now have new generated server and client code, but we still need to implement
+and call the new method in the human-written parts of our example application.
+
+### Update the server
+
+Edit `greeter_server/main.go` and add the foloowing function to it:
```
Comment (menghanl) [inline docs/quickstart/go.md]: s/foloowing/following

File: docs/quickstart/go.md [hunk 7]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+
+```
```
Comment (menghanl) [inline docs/quickstart/go.md]: Add `go` at the end to have the code highlighted with go syntax.

File: docs/quickstart/go.md [hunk 8]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+func (s *server) SayHelloAgain(ctx context.Context, in *pb.HelloRequest) (*pb.HelloReply, error) {
+        return &pb.HelloReply{Message: "Hello again " + in.Name}, nil
+}
+```
+
+### Update the client
+
+Edit `greeter_client/main.go` to add the following code to the main function.
+
+```
```
Comment (menghanl) [inline docs/quickstart/go.md]: `go` syntax highlight here, too.

File: docs/quickstart/go.md [hunk 9]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+if err != nil {
+        log.Fatalf("could not greet: %v", err)
+}
+log.Printf("Greeting: %s", r.Message)
+```
+
+### Run!
+
+1. Run the server 
+   ``` $ go run greeter_server/main.go
```
Comment (menghanl) [inline docs/quickstart/go.md]: Add back-quotes at the end.

File: docs/quickstart/go.md [hunk 10]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+
+2. On a different terminal, run the Client
```
Comment (menghanl) [inline docs/quickstart/go.md]: In the rendered view, the next line is inlined. Probably need a blank line here...

File: docs/quickstart/go.md [hunk 11]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+
+1. Run the server 
+   ``` $ go run greeter_server/main.go
+
+2. On a different terminal, run the Client 
+   ``` $ go run greeter_client/main.go ```
+
+You should see the updated output:
+   ```
+   $ ./greeter_client
```
Comment (menghanl) [inline docs/quickstart/go.md]: Remote this command here. We don't have it.

File: docs/quickstart/go.md [hunk 12]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+## Try it!
+Now to compile and run the server and client apps use go run. Make sure you're still in the same examples directory.
+
+```$ go run greeter_server/main.go```
+
+From a different terminal run the following:
+
+```$ go run greeter_client/main.go```
+
+If things go smoothly, you will see the `Greeter received: Hello world` in the client side output.
```
Comment (menghanl) [inline docs/quickstart/go.md]: The output should be something like `Greeting: Hello world`...
The message printed in out example is different...

File: docs/quickstart/go.md [hunk 13]
```diff
@@ -4,4 +4,184 @@ headline: Go Quick Start
...
+1. Run the server 
+   ``` $ go run greeter_server/main.go
+
+2. On a different terminal, run the Client 
+   ``` $ go run greeter_client/main.go ```
+
+You should see the updated output:
+   ```
+   $ ./greeter_client
+   Greeter received: Hello world
```
Comment (menghanl) [inline docs/quickstart/go.md]: Output message should be `Greeting: Hello world`.

File: docs/quickstart/go.md (hunk 4 above)
Comment (MakMukhi) [inline docs/quickstart/go.md]: I realize that but I thought the users might want to know about how to actually generate that file. I can remove that if you find it extraneous.

File: docs/quickstart/go.md (hunk 5 above)
Comment (MakMukhi) [inline docs/quickstart/go.md]: I got that part from the c++ quick start. Removing it.

File: docs/quickstart/go.md [hunk 14]
```diff
@@ -4,4 +4,186 @@ headline: Go Quick Start
...
+### Prerequisites
+
+#### Go version
+Make sure that Go installed is version 1.5 or higher.
+
+```sh
+$ go version
+```
+
+For installtion instruction follow this guide: [Getting Started - The Go Programming Language](https://golang.org/doc/install)
```
Comment (LisaFC) [inline docs/quickstart/go.md]: Typo: installtion -> installation

File: docs/quickstart/go.md [hunk 15]
```diff
@@ -4,4 +4,186 @@ headline: Go Quick Start
...
+#### Install gRPC
+To install gRPC on your system use the following command:
+```sh
+$ go get google.golang.org/grpc
+```
+
+#### Install Protocol Buffers v3
+
+While not mandatory, gRPC usually leverages Protocol Buffers v3 for service
+definitions and data serialization.
```
Comment (LisaFC) [inline docs/quickstart/go.md]: This sentence is a bit wordy/superfluous for this quickstart - users can find out more about how/if gRPC uses protocol buffers in the more detailed guides and tutorials. I'd take it out.

File: docs/quickstart/go.md [hunk 16]
```diff
@@ -4,4 +4,186 @@ headline: Go Quick Start
...
+
+<div id="toc"></div>
+
+## Before you begin
+
+### Prerequisites
+
+#### Go version
+Make sure that Go installed is version 1.5 or higher.
+
```
Comment (hsaliak) [inline docs/quickstart/go.md]: Simplify to
"gRPC works with Go 1.5 or higher."

File: docs/quickstart/go.md (hunk 14 above)
Comment (hsaliak) [inline docs/quickstart/go.md]: For installation instructions, follow this guide: [ ..]

File: docs/quickstart/go.md [hunk 17]
```diff
@@ -4,4 +4,186 @@ headline: Go Quick Start
...
+```sh
+$ go version
+```
+
+For installtion instruction follow this guide: [Getting Started - The Go Programming Language](https://golang.org/doc/install)
+
+#### Install gRPC
+To install gRPC on your system use the following command:
```
Comment (hsaliak) [inline docs/quickstart/go.md]: Use the following command to install gRPC

File: docs/quickstart/go.md [hunk 18]
```diff
@@ -4,4 +4,186 @@ headline: Go Quick Start
...
+
+## Download the example
+The grpc code that was fetched with ``` $ go get google.golang.org/grpc ``` also contains the examples. They can be found under the examples dir: `$GOPATH/src/google.golang.org/grpc/examples`.
+
+## Build the example
+Change to the example directory
+
+```$ cd $GOPATH/src/google.golang.org/grpc/examples/helloworld```
+
+For gRPC services to work we want to compile the proto file to generate a pb.go file that contains the generated code which is then used by the server and client apps. This file is already generated and shipped with the example code and can be found under the directory: `$GOPATH/src/google.golang.org/grpc/examples/helloworld/helloworld`
```
Comment (hsaliak) [inline docs/quickstart/go.md]: gRPC services are defined in a proto file, which is used to generate a corresponding .pb.go file that contains generated code. This file is already generated for the  ```helloworld``` example code and can be found ..

File: docs/quickstart/go.md [hunk 19]
```diff
@@ -4,4 +4,186 @@ headline: Go Quick Start
...
+
+This `helloworld.pb.go` file contains our generated client and server code, as well as code for populating, serializing, and retrieving our `HelloRequest` and `HelloResponse` message types.
+
+
+## Try it!
+Now to compile and run the server and client apps use go run. Make sure you're still in the same examples directory.
```
Comment (hsaliak) [inline docs/quickstart/go.md]: Technically go run does not leave behind a compiled binary.. this may be odd. 
Should we break it down into go build and go run steps?

If we want to stick to just using `go run`, I recommend the following changes, 
To compile and run the server and client code, the `go run`` command can be used. 
In the examples directory:

**Comment by hsaliak** [comment]
LGTM

**Comment by menghanl** [comment]
LGTM.
